### PR TITLE
remove direct dependency on DiffResults

### LINKED ISSUE #90: Something I noticed in #88: Is there a reason to keep DiffResults as a proper dependency even though it is hidden with `@require`?
Something I noticed in #88: Is there a reason to keep DiffResults as a proper dependency even though it is hidden with `@require`?

I think one could either remove the `@require` around `DiffResults_helpers.jl` or make it a test-only dependency (if one wants to keep the compat entries, otherwise it could be removed completely from Project.toml).

_Originally posted by @devmotion in https://github.com/tpapp/LogDensityProblems.jl/pull/89#discussion_r954791744_

### PR DESCRIPTION
**Pull request by tpapp**
Fixes #90.

### CODE REVIEW

File: src/AD_ForwardDiff.jl [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 import .ForwardDiff
 
+import DiffResults # should load DiffResults_helpers.jl
+
```
Comment (devmotion) [inline src/AD_ForwardDiff.jl:8]: I wonder if this is needed? ForwardDiff already loads DiffResults, so shouldn't the helpers be loaded automatically?

Comment (tpapp) [reply src/AD_ForwardDiff.jl:8]: I am wondering myself, but technically ForwardDiff *could* remove DiffResults as a dependency (make it conditional etc) and then we would have a problem.

My impression is that this is innocuous, so I would leave it in.

Comment (devmotion) [reply src/AD_ForwardDiff.jl:8]: True, it seems a bit safer to keep it.

Comment (devmotion) [reply src/AD_ForwardDiff.jl:8]: However, does this actually work? DiffResults is not a direct dependency, so wouldn't we need something like
```suggestion
import ForwardDiff.DiffResults # should load DiffResults_helpers.jl

```
?

Comment (devmotion) [reply src/AD_ForwardDiff.jl:8]: Ah no, I see ForwardDiff exports it:
https://github.com/JuliaDiff/ForwardDiff.jl/blob/6a6443b754b0fcfb4d671c9a3d01776df801f498/src/ForwardDiff.jl#L28

Comment (devmotion) [reply src/AD_ForwardDiff.jl:8]: Ah but again, we don't load ForwardDiff but just import it, so I guess we should still use
```suggestion
import ForwardDiff.DiffResults # should load DiffResults_helpers.jl

```
or something similar

File: src/AD_ForwardDiff.jl [hunk 2]
```diff
@@ -4,6 +4,8 @@
 
 import .ForwardDiff
```
Comment (devmotion) [inline src/AD_ForwardDiff.jl:5]: I already assumed that Requires chokes with `using ForwardDiff`/`import ForwardDiff`. I guess the following works:
```suggestion
import .ForwardDiff: ForwardDiff, DiffResults
```

**Comment by tpapp** [comment]
@devmotion: your original suggestion works with the extra `.`; I am not sure which to prefer.

**Review by devmotion — APPROVED** [review]
Looks good to me.

**Pull request merged**